blaze/expr/datetime.py

```diff
@@ -119,6 +129,27 @@ def utcfromtimestamp(expr):
     return UTCFromTimestamp(expr)
 
 
+class nanosecond(DateTime): _dtype = datashape.int64
```

Comment:
This is probably getting annoying by now, but we need to think about Option types and datetimes, so I think we need to do something like:

```python
class week(DateTime):
    @property
    def _dtype(self):
        return datashape.int64 if self._child._dtype == datashape.datetime_ else Option(datashape.datetime_)
```

And similar for the `string` and `bool_` `_dtype`s here.